From: Peter Korsgaard <jacmet@sunsite.dk>
Subject: [PATCH] fix json_reformat linking on uClibc

Upstream-Status: Pending

json_reformat calls yajl_gen_* functions, which internally use isnan() /
isinf(). Link with -lm.

Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
---
--- a/reformatter/CMakeLists.txt	2026-06-29 04:20:36.165687101 +0000
+++ b/reformatter/CMakeLists.txt	2026-06-29 04:20:36.305685112 +0000
@@ -26,7 +26,7 @@
 
 ADD_EXECUTABLE(json_reformat ${SRCS})
 
-TARGET_LINK_LIBRARIES(json_reformat yajl_s)
+TARGET_LINK_LIBRARIES(json_reformat yajl_s m)
 
 # In some environments, we must explicitly link libm (like qnx,
 # thanks @shahbag)
--- a/perf/CMakeLists.txt	2026-06-29 04:20:34.421711868 +0000
+++ b/perf/CMakeLists.txt	2026-06-29 04:20:36.305685112 +0000
@@ -20,4 +20,4 @@
 
 ADD_EXECUTABLE(perftest ${SRCS})
 
-TARGET_LINK_LIBRARIES(perftest yajl_s)
+TARGET_LINK_LIBRARIES(perftest yajl_s m)
--- a/example/CMakeLists.txt	2026-06-29 04:20:34.421711868 +0000
+++ b/example/CMakeLists.txt	2026-06-29 04:20:36.309685056 +0000
@@ -20,4 +20,4 @@
 
 ADD_EXECUTABLE(parse_config ${SRCS})
 
-TARGET_LINK_LIBRARIES(parse_config yajl_s)
+TARGET_LINK_LIBRARIES(parse_config yajl_s m)
--- a/test/api/CMakeLists.txt	2026-06-29 04:20:34.421711868 +0000
+++ b/test/api/CMakeLists.txt	2026-06-29 04:20:36.309685056 +0000
@@ -21,5 +21,5 @@
 FOREACH (test ${TESTS})
   GET_FILENAME_COMPONENT(testProg ${test} NAME_WE)
   ADD_EXECUTABLE(${testProg} ${test})
-  TARGET_LINK_LIBRARIES(${testProg} yajl)
+  TARGET_LINK_LIBRARIES(${testProg} yajl m)
 ENDFOREACH()
--- a/test/parsing/CMakeLists.txt	2026-06-29 04:20:34.421711868 +0000
+++ b/test/parsing/CMakeLists.txt	2026-06-29 04:20:36.313684999 +0000
@@ -20,4 +20,4 @@
 
 ADD_EXECUTABLE(yajl_test ${SRCS})
 
-TARGET_LINK_LIBRARIES(yajl_test yajl_s)
+TARGET_LINK_LIBRARIES(yajl_test yajl_s m)
--- a/verify/CMakeLists.txt	2026-06-29 04:20:36.173686986 +0000
+++ b/verify/CMakeLists.txt	2026-06-29 04:20:36.317684943 +0000
@@ -26,7 +26,7 @@
 
 ADD_EXECUTABLE(json_verify ${SRCS})
 
-TARGET_LINK_LIBRARIES(json_verify yajl_s)
+TARGET_LINK_LIBRARIES(json_verify yajl_s m)
 
 # copy in the binary
 ADD_CUSTOM_COMMAND(TARGET json_verify POST_BUILD
